test/framework/universal_app.go

```diff
@@ -193,36 +196,196 @@ type UniversalApp struct {
 	container    string
 	ip           string
 	verbose      bool
+	dockerOpts   *docker.RunOptions
 }
 
-func NewUniversalApp(t testing.TestingT, clusterName, dpName string, mode AppMode, isipv6, verbose bool, caps []string) (*UniversalApp, error) {
-	app := &UniversalApp{
-		t:            t,
-		ports:        map[string]string{},
+type UniversalAppOpts struct {
```

Comment:
I wonder if we really need to introduce this type. Could we just use functional options with `NewUniversalApp`, e.g.

```Go
app, err := NewUniversalApp(
      AppTestingOpt(t),
      AppVerboseOpt(true),
      AppEnableIPv6Opt(false),

```